### ISSUE TITLE
Add guards to index assignment in Universe creation

### ISSUE THREAD

**Issue by yuxuanzhuang** [MDAnalysis/mdanalysis#5223]
## Expected behavior ##

A clear error message should be raised when an incorrect index assignment is provided to `Universe.empty`, e.g.

```python
u_new = mda.Universe.empty(
            n_atoms=2,
            n_residues=2, atom_resindex=[1,2],
            trajectory=True
            )
u_new.add_TopologyAttr("resnames", ['RES1', 'RES2'])
```
This is incorrect because `atom_resindex` should be 0-based. (And I confused it with resids when I encountered this error).
## Actual behavior ##

No error is raised during creation but only when accessing it 

```u_new.atoms.resnames``` provides `IndexError: index 2 is out of bounds for axis 0 with size 2`

This is also true for `residue_segindex`.
## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`): dev
- Which version of Python (`python -V`)? 3.12
- Which operating system? mac-os

**Comment by Akshat-Raj** [comment]
Hi, I'm interested in working on this issue, raising a draft pr now!
